lib/common/operations.c

```diff
@@ -544,3 +544,33 @@ pcmk_xe_is_probe(xmlNode *xml_op)
     pcmk__scan_min_int(interval_ms_s, &interval_ms, 0);
     return pcmk_is_probe(task, interval_ms);
 }
+
+bool
+pcmk_xe_mask_probe_failure(xmlNode *xml_op)
+{
+    int status = PCMK_EXEC_UNKNOWN;
+    int rc = PCMK_OCF_OK;
+
+    if (!pcmk_xe_is_probe(xml_op)) {
+        return false;
+    }
+
+    crm_element_value_int(xml_op, XML_LRM_ATTR_RC, &rc);
+    crm_element_value_int(xml_op, XML_LRM_ATTR_OPSTATUS, &status);
+
+    switch (status) {
+        case PCMK_EXEC_DONE:
+            return (rc == PCMK_OCF_NOT_INSTALLED || rc == PCMK_OCF_NOT_CONFIGURED);
+
+        case PCMK_EXEC_NOT_INSTALLED:
+            return rc == PCMK_OCF_NOT_INSTALLED;
+
+        case PCMK_EXEC_ERROR:
+        case PCMK_EXEC_ERROR_HARD:
+        case PCMK_EXEC_ERROR_FATAL:
+            return (rc == PCMK_OCF_NOT_INSTALLED || rc == PCMK_OCF_NOT_CONFIGURED);
+
+        default:
+            return false;
+    }
```

Comment:
Looking at this again, I'm thinking we can simplify it as

    return rc == PCMK_OCF_NOT_INSTALLED || rc == PCMK_OCF_NOT_CONFIGURED || status == PCMK_EXEC_NOT_INSTALLED

Any of those codes should be sufficient by themselves

Reply:
Fixed.